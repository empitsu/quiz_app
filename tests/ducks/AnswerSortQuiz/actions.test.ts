import {
  popFromSelectedOptions,
  pushToSelectedOptions,
  reset,
} from "../../../src/ducks/AnswerSortQuiz";

const testOption = {
  optionId: 1,
  text: "test",
  originalIndex: 1,
  selected: false,
};
describe("Actions of AnswerSortQuiz", () => {
  describe("popFromSelectedOptions()", () => {
    it("should create an action to remove from selected options", () => {
      expect(popFromSelectedOptions(testOption)).toMatchSnapshot();
    });
  });
  describe("pushToSelectedOptions()", () => {
    it("should create an action to push to selected options", () => {
      expect(pushToSelectedOptions(testOption)).toMatchSnapshot();
    });
  });
  describe("reset()", () => {
    it("should create an action to reset", () => {
      expect(reset([testOption])).toMatchSnapshot();
    });
  });
});
