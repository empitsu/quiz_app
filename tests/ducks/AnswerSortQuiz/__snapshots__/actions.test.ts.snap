// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions of AnswerSortQuiz popFromSelectedOptions() should create an action to remove from selected options 1`] = `
Object {
  "payload": Object {
    "optionId": 1,
    "originalIndex": 1,
    "selected": false,
    "text": "test",
  },
  "type": "POP_FROM_SELECTED_OPTIONS",
}
`;

exports[`Actions of AnswerSortQuiz pushToSelectedOptions() should create an action to push to selected options 1`] = `
Object {
  "payload": Object {
    "optionId": 1,
    "originalIndex": 1,
    "selected": false,
    "text": "test",
  },
  "type": "PUSH_T0_SELECTED_OPTIONS",
}
`;

exports[`Actions of AnswerSortQuiz reset() should create an action to reset 1`] = `
Object {
  "payload": Array [
    Object {
      "optionId": 1,
      "originalIndex": 1,
      "selected": false,
      "text": "test",
    },
  ],
  "type": "RESET",
}
`;
